pumps/elasticsearch.go

```diff
@@ -71,6 +72,16 @@ type ElasticsearchConf struct {
 	Username string `json:"auth_basic_username" mapstructure:"auth_basic_username"`
 	// Basic auth password. It's send to ES in the Authorization header as username:password encoded in base64.
 	Password string `json:"auth_basic_password" mapstructure:"auth_basic_password"`
+	// Enables SSL connection.
+	UseSSL bool `json:"use_ssl" mapstructure:"use_ssl"`
+	// Controls whether the pump client verifies the Elastic Search server's certificate chain and hostname.
+	SSLInsecureSkipVerify bool `json:"ssl_insecure_skip_verify" mapstructure:"ssl_insecure_skip_verify"`
+	// Can be used to set custom certificate file for authentication with Elastic Search.
+	SSLCertFile string `json:"ssl_cert_file" mapstructure:"ssl_cert_file"`
+	// Can be used to set custom key file for authentication with Elastic Search.
+	SSLKeyFile string `json:"ssl_key_file" mapstructure:"ssl_key_file"`
+
+	tlsConfig *tls.Config
```

Comment:
I would remove this field, as the value of this field depends on the others SSL fields then I don't think we need it. And instead build a function that returns the TLSConfig based on current configuration. Also consider that everytime that we add new fields in this kind of struct we need to set a json tag, Eg: `json:"-"` or `json:"tls_config"`

Reply:
Understood - I'm removing it!

I didn't set the json tag as it's something that we're setting during the program execution, not from an env var. But makes totally sense to delete it. Thanks!